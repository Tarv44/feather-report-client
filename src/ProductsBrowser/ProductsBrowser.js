import React, {Component} from 'react';
import styles from './ProductsBrowser.module.css';
import ProductContext from '../productContext';
import {companies, categories, prodWithFeat} from '../dummy-store';
import CatSelector from '../CatSelector/CatSelector';
import ProductsSection from '../ProductsSection/ProductsSection';
import { NavLink } from 'react-router-dom';

export default class ProductsBrowser extends Component {
    state = {
        company: {},
        categories: [],
        catFilter: '',
        products: []
    }

    static contextType = ProductContext

    componentDidMount() {
        const companyNameId = this.props.match.params.co_path

        const products = prodWithFeat.map(p => {
            const isSelected = this.context.selected.findIndex(s => s.id === p.id) >= 0
            if (isSelected) {
                p.selected = true
            } else {
                p.selected = false
            }
            return p
        })



        this.setState({ 
            company: companies[companyNameId],
            categories,
            products
        })
    }

    updateCatFilter = (catFilter) => {
        this.setState({ catFilter })
    }

    updateSelected = (id) => {
        const index = this.state.products.findIndex(p => p.id === id)
        const products = this.state.products
        products[index].selected = !products[index].selected
        this.setState({ products })
    }

    render() {
        return (
            <main>
                <header>
                    <h1>{this.state.company.name}</h1>
                    <NavLink to={'/compare'}>Compare</NavLink>
                    <p>Selected: {this.context.selected.length}/4</p>
                </header>
                <main className={styles.browser_main}>
                    <CatSelector 
                        categories={this.state.categories}
                        handleFilter={this.updateCatFilter}
                    />
                    <ProductsSection 
                        products={this.state.products} 
                        type={'select'}
                        catFilter={this.state.catFilter}
                        handleSelected={this.updateSelected} 
                    /> 
                </main>
            </main>
        )
    }
}
