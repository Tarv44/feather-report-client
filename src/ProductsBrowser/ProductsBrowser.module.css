.browser_main {
    display: flex;
    flex-direction: column;
}

.float {
    position: fixed;
    bottom: 0;
    right: 0;
    margin-right: 1em;
    margin-bottom: .5em;
    display: flex;
    flex-direction: column;
    align-items: flex-end;
    background: grey;
    border: solid black 1px;
    padding: 4px;
    box-shadow: 0 3px 10px 2px grey;
}

.float p {
    margin: 0;
    margin-bottom: .25em;
    color: white;
}

.float button {
    font-size: 18px;
}

@media (min-width: 1280px) {
    .browser_main {
        flex-direction: row;
    }
}