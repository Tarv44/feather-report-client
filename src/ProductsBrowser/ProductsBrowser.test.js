import React from 'react'
import ReactDom from 'react-dom'
import renderer from 'react-test-renderer'
import ProductsBrowser from './ProductsBrowser'
import { products } from '../component.fixtures'
import ProductContext from '../productContext'
import { BrowserRouter } from 'react-router-dom'

const selected = products.slice(3)
const contextValue = { selected }
const match = {
    params: {
        co_path: '/path'
    }
}

describe('----- ProductsBrowser -----', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDom.render(
            <ProductContext.Provider value={contextValue}>
                <BrowserRouter>
                    <ProductsBrowser match={match}/>
                </BrowserRouter> 
            </ProductContext.Provider>    
        , div)
        ReactDom.unmountComponentAtNode(div);
    })
    it('renders the UI as expected', () => {
        const comp = renderer
            .create(
                <ProductContext.Provider value={contextValue}>
                    <BrowserRouter>
                        <ProductsBrowser match={match}/>
                    </BrowserRouter> 
                </ProductContext.Provider>  
            )
            .toJSON()
        expect(comp).toMatchSnapshot()
    })
})