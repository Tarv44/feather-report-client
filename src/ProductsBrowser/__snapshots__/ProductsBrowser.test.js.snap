// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`----- ProductsBrowser ----- renders the UI as expected 1`] = `
<main>
  <header>
    <h1 />
  </header>
  <main
    className="browser_main"
  >
    <section
      className="section"
    >
      <h2>
        Categories
      </h2>
      <div
        className="buttons"
      >
        <button
          onClick={[Function]}
        >
          All Categories
        </button>
      </div>
    </section>
    <section
      className="section"
    />
    <div
      className="float"
    >
      <p>
        11
        /4
      </p>
      <a
        aria-current={null}
        href="/compare"
        onClick={[Function]}
      >
        <button>
          Compare
        </button>
      </a>
    </div>
  </main>
</main>
`;
